core/prover/src/client.rs

```diff
@@ -0,0 +1,223 @@
+            Ok(v) => v,
+            Err(e) => {
+                return Err(format!("register request failed: {}", e));
+            }
+        };
+        let text = match res.text() {
+            Ok(v) => v,
+            Err(e) => {
+                return Err(format!("failed to read register response: {}", e));
+            }
+        };
+        match i32::from_str(&text) {
+            Ok(v) => Ok(v),
+            Err(e) => Err(format!("failed to parse register prover id: {}", e)),
+        }
+    }
+
+    pub fn prover_stopped(&self, prover_run_id: i32) -> Result<(), String> {
+        let client = reqwest::Client::new();
+        let res = client.post(&self.stopped_url).json(&prover_run_id).send();
+        let res = match res {
+            Ok(v) => v,
+            Err(e) => {
+                return Err(format!("prover stopped request failed: {}", e));
+            }
+        };
+        if res.status() != reqwest::StatusCode::OK {
+            return Err(format!(
+                "prover stopped request failed with status: {}",
+                res.status()
+            ));
+        }
+        Ok(())
+    }
+}
+
+impl crate::ApiClient for ApiClient {
+    fn block_to_prove(&self) -> Result<Option<(i64, i32)>, String> {
+        // TODO: handle errors
+        let client = reqwest::Client::new();
+        let res = client
+            .get(&self.block_to_prove_url)
+            .json(&server::ProverReq {
+                name: self.worker.clone(),
+            })
+            .send();
+        let mut res = match res {
+            Ok(v) => v,
+            Err(e) => {
+                return Err(format!("block to prove request failed: {}", e));
+            }
+        };
+        let text = match res.text() {
+            Ok(v) => v,
+            Err(e) => {
+                return Err(format!("failed to read block to prove response: {}", e));
+            }
+        };
+        let res: server::BlockToProveRes = serde_json::from_str(&text).unwrap();
+        if res.block != 0 {
+            return Ok(Some((res.block, res.prover_run_id)));
+        }
+        Ok(None)
+    }
+
+    fn working_on(&self, job_id: i32) -> Result<(), String> {
+        // TODO: handle errors
+        let client = reqwest::Client::new();
+        let res = client
+            .post(&self.working_on_url)
+            .json(&server::WorkingOnReq {
+                prover_run_id: job_id,
+            })
+            .send();
+        let res = match res {
+            Ok(v) => v,
+            Err(e) => {
+                return Err(format!("failed to send working on request: {}", e));
+            }
+        };
+        if res.status() != reqwest::StatusCode::OK {
+            return Err(format!(
+                "working on request failed with status: {}",
+                res.status()
+            ));
+        }
+        Ok(())
+    }
+
+    fn prover_data(&self, block: i64, timeout: time::Duration) -> Result<ProverData, String> {
```

Comment:
I know. That's a problem to be addressed separately. Storage crate functions accept `i64` as a block number in one place, and `i64` in other. I'm using i64, because I use accept i64.